pkg/terminal/command.go

```diff
@@ -829,7 +819,7 @@ func threads(t *Term, ctx callContext, args string) error {
 	if err != nil {
 		return err
 	}
-	sort.Sort(byThreadID(threads))
+	slices.SortFunc(threads, func(a, b *api.Thread) int { return cmp.Compare(a.ID, b.ID) })
```

Comment:
`return a.ID - b.ID`?

Reply:
We can't change to `return int(a.ID - b.ID)` because it will incorrectly sort values:

The following program demonstrates that:
```go
package main

import (
	"cmp"
	"fmt"
	"math"
	"slices"
)

type constantValue struct {
	value int64
}

func main() {
	valuesSub := []constantValue{{math.MaxInt64}, {1}, {math.MinInt64}}
	valuesCmp := slices.Clone(valuesSub)
	slices.SortFunc(valuesSub, func(a, b constantValue) int { return int(a.value - b.value) })
	fmt.Println("Sorted valuesSub:", valuesSub) // Sorted valuesSub: [{1} {9223372036854775807} {-9223372036854775808}]

	slices.SortFunc(valuesCmp, func(a, b constantValue) int { return cmp.Compare(a.value, b.value) })
	fmt.Println("Sorted valuesCmp:", valuesCmp) // Sorted valuesCmp: [{-9223372036854775808} {1} {9223372036854775807}]

}
```

Reply:
https://github.com/go-delve/delve/pull/3880#discussion_r1893880698